lib/vanagon/platform/solaris_10.rb

```diff
@@ -10,6 +10,19 @@ def generate_package(project)
         name_and_version = "#{project.name}-#{project.version}"
         pkg_name = package_name(project)
 
+        # Generate list of dirs in the package and create an exlplicit search
+        # string for AWK to use in order to explicitly define which directories
+        # to put in to the prototype file. Note that the Regexp object was avoided
+        # because the output from Regexp would create failures from AWK as the conversion
+        # from a Regexp obj to a string is formatted in a sub-optimal way that would have
+        # required more string manipulation anyway. the string should be formatted like so:
+        #         && ($$3 ~ /directory\/regex.*/ || $$3 ~ /another\/directory\/regex.*/)
+        # for as many iterations as there are directries in the package
+        pkgdirs = project.get_root_directories.map { |dir| dir.sub(/^\//, "").gsub(/[\/]/, "\\/").gsub(/\./, "\\.") + '.*' }
```

Comment:
Also, I think the `sub(/^\//, '')` pattern appears often enough in vanagon it would be worth tossing into https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/extensions/string.rb